services/stake.tsx

```diff
@@ -48,15 +55,21 @@ class StakeContract {
       HOST_ABI.abi,
       signer
     ) as AllocateStakingToken;
-    return new StakeContract(contract, wallet);
+    return new StakeContract(contract, wallet, provider);
+  }
+
+  fetchStakes(address = this.wallet) {
+    return this.contract.fetchStakes(address);
   }
 
-  async fetchStakes() {
-    return this.contract.fetchStakes(this.wallet);
+  fetchStakerDetails(address: string) {
+    return this.contract.fetchStakerDetails(address);
   }
 
   async createStake(amount: number, duration: IStakeDuration) {
-    return this.contract.createStake(amount, duration);
+    const tokens = ethers.utils.parseUnits(amount.toString(), 18);
+    await approveUSDCX(this.provider, this.wallet, amount, CONTRACT_ADDRESS);
+    return this.contract.createStake(tokens, duration);
```

Comment:
This is failing with error "execution reverted: SuperfluidToken: move amount exceeds balance"